packages/flutter/lib/src/material/focusable.dart

```diff
@@ -0,0 +1,297 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+typedef FocusableOnKeyCallback = bool Function(FocusableNode node, RawKeyEvent event);
+
+/// A widget that manages a [FocusableNode] and allows specification of the
+/// focus order.
+///
+/// The node can be either a scope node or a leaf node. Scope nodes provide a
+/// scope for their children, using the focus traversal policy defined by the
+/// [DefaultFocusTraversal] widget above them to traverse their children.
+///
+/// It provides [onFocusChange] as a way to be notified when the focus is given
+/// to or removed from this widget, and allows specification of a [focusedDecoration]
+/// to be shown when its [child] has focus.
+///
+/// The [onKey] argument allows specification of a key even handler that should
+/// be invoked when this node or one of its children has focus.
+class Focusable extends StatefulWidget {
+  /// Creates a widget that manages a [FocusableNode]
+  ///
+  /// The [child] arguments is required and must not be null.
+  ///
+  /// The [isScope], [autofocus], and [showDecorations] arguments must not be
+  /// null.
+  const Focusable({
+    Key key,
+    @required this.child,
+    this.isScope = false,
+    this.autofocus = false,
+    this.showDecorations = true,
+    this.focusedDecoration,
+    this.unfocusedDecoration,
+    this.onFocusChange,
+    this.onKey,
+    this.debugLabel,
+  })  : assert(child != null),
+        assert(isScope != null),
+        assert(autofocus != null),
+        assert(showDecorations != null),
+        super(key: key);
+
+  /// A debug label for this widget.
+  final String debugLabel;
+
+  /// The child widget of this [Focusable].
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// Handler for keys pressed when this object or one of its children has
+  /// focus.
+  ///
+  /// Key events are first given to the leaf nodes, and if they don't handle
+  /// them, then to each node up the widget hierarchy. If they reach the root of
+  /// the hierarchy, they are discarded.
+  final FocusableOnKeyCallback onKey;
```

Comment:
probably mention there that this isn't what you want to handle text input (e.g. for text fields)

Reply:
Done.